typesafety/test_issue_49.yml

```diff
@@ -7,3 +7,51 @@
     logger.debug(123.3)
     logger.trace(1)
     logger.error(.3)
+- case: variables
+  parametrized:
+    - level: info
+    - level: debug
+    - level: warning
+    - level: error
+    - level: trace
+    - level: exception
+  main: |
+    from loguru import logger
+
+    foo = "bar"
+    logger.{{ level }}(foo)
+- case: variables__type-checked__wrong
+  main: |
+    from loguru import logger
+    from typing_extensions import Final
+
+    foo: Final = "bar"
+    logger.info(foo, 1)
+  out: |
+    main:5: error: Not all arguments converted during string formatting  [str-format]
+- case: variables__type-checked__correct
+  main: |
+    from loguru import logger
+    from typing_extensions import Final
+
+    foo: Final = "bar {}"
+    logger.info(foo, 1)
+- case: variables__mypy_cant_tell
+  main: |
+    from loguru import logger
+    from typing_extensions import Final
+
+    foo = "bar"
```

Comment:
@sobolevn tis the case I would love get help with.

Reply:
I would love to help! What is the context? 🤔

Reply:
@sobolevn I believe [this](https://github.com/kornicameister/loguru-mypy/pull/80/files#r551645588) comment will make up for the context. @ThibaultLemaire  did awesome job spotting various things that might be helpful here.

Reply:
Done! 👍